TicTacToeApp/ComputerPlayer.swift

```diff
@@ -16,10 +26,48 @@ public class ComputerPlayer {
       }
     }
 
-    return getRandomElement(availableSpaces)
+    return availableSpaces
   }
 
   private func getRandomElement(array: [Int]) -> Int {
     return array[Int(arc4random_uniform(UInt32(array.count)))]
   }
+
+  private func makeBlockingRequest(request: NSURLRequest, onSuccess: (String) -> Void) {
+    let semaphore = dispatch_semaphore_create(0)
+
+    NSURLSession.sharedSession().dataTaskWithRequest(request) { (data, response, error) -> Void in
+      guard error == nil && data != nil else {
+        dispatch_semaphore_signal(semaphore)
+        return
+      }
+
+      let reply = String(data: data!, encoding: NSUTF8StringEncoding)
+      dispatch_semaphore_signal(semaphore)
+
+      onSuccess(reply!)
+    }.resume()
+    
+    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
+  }
+
+  private func computerMoveRequest(gameState: Game, baseUrl: String) -> NSURLRequest {
+    let currentPlayerParam = "current_player=" + markToString(gameState.getCurrentPlayer())
+    let boardParam = "board=" + boardToString(gameState.getBoardMarks())
+    let url = baseUrl + "?" + currentPlayerParam + "&" + boardParam
+
+    return NSURLRequest(URL: NSURL(string: url)!)
+  }
+
+  private func boardToString(board: [PlayerMark?]) -> String {
+    return board.map{ markToString($0!) }.joinWithSeparator(",")
+  }
+
+  private func markToString(mark: PlayerMark) -> String {
+    switch (mark) {
+    case PlayerMark.NONE: return "_"
```

Comment:
I think you can just do `case .NONE`. I believe the compiler is smart enough to infer that it is a `PlayerMark`.